using_restricted_array.rb

```diff
@@ -7,41 +7,236 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  # if array is empty done
+  # set i to 0
+  # while i is less than array length
+  # - increment i by 1
+  # outside loop, return i
+  i = 0
+  while array[i] != nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  # we know the length of the array
+  # it contains integers
+  # want: print each integer value in array
+
+  # set variable to array's length
+  # loop array by array's length times
+  # - print value at each index
+
+
+
+  length(array).times do |value|
+    print "#{array[value]}, "
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+=begin
+1. We know length of array
+2. it contains integets
+3. it is unsorted
+
+Want: find a known value in array
+once found return true
+return false if not found.
+
+Algorithm
+1. set i to 0
+2. while i < length
+  - check value at i
+  - if a match return true
+  - increment i by 1
+3. outside loop return false
+=end
+
+  i = 0
+
+  while i < length
+    if array[i] == value_to_find
+
+      return true
+    end
+    i += 1
+    return false
```

Comment:
Currently, if the value at array[0] is not the value you're looking for, you'll return false. Which is incorrect.

The return false statement should be outside the loop.